frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikValues, useFormikContext } from 'formik';
+import { TextInputTypes, InputGroupText } from '@patternfly/react-core';
+import { InputGroupField, InputField, DropdownField, ActionGroupWithIcons } from '@console/shared';
+import {
+  HTTPRequestTypeForm,
+  TCPRequestTypeForm,
+  CommandRequestTypeForm,
+} from './RequestTypeForms';
+import { RequestType } from './health-checks-types';
+import FormSection from '../import/section/FormSection';
+import './ProbeForm.scss';
+
+const getRequestTypeForm = (
+  value: string,
+  probeType: string,
+  ports?: { [port: number]: number },
+) => {
+  switch (value) {
+    case RequestType.HTTPGET:
+      return <HTTPRequestTypeForm ports={ports} probeType={probeType} />;
+    case RequestType.ContainerCommand:
+      return <CommandRequestTypeForm probeType={probeType} />;
+    case RequestType.TCPSocket:
+      return <TCPRequestTypeForm ports={ports} probeType={probeType} />;
+    default:
+      return null;
+  }
+};
+
+interface ProbeFormProps {
+  containerPorts?: { [port: number]: number };
+  onSubmit: () => void;
+  onClose: () => void;
+  probeType: string;
+}
+
+enum RequestTypeOptions {
+  httpGet = 'HTTP GET',
+  command = 'Container Command',
+  tcpSocket = 'TCP Socket',
+}
+
+const ProbeForm: React.FC<ProbeFormProps> = ({ onSubmit, onClose, containerPorts, probeType }) => {
+  const {
+    values: { healthChecks },
+    setFieldValue,
+    errors,
+  } = useFormikContext<FormikValues>();
+  const onRequestTypeChange = (value) => {
+    setFieldValue(`healthChecks.${probeType}.data.requestType`, value);
+  };
+  return (
+    <div className="odc-heath-check-probe-form">
+      <FormSection>
+        <InputField
+          type={TextInputTypes.number}
+          name={`healthChecks.${probeType}.data.failureThreshold`}
+          label="Failure Threshold"
+          style={{ maxWidth: '100%' }}
+          helpText={'How many times the probe will try starting or restarting before giving up.'}
+        />
+        <DropdownField
+          name={`healthChecks.${probeType}.data.requestType`}
+          label="Type"
+          items={RequestTypeOptions}
+          title={RequestType.HTTPGET}
+          onChange={onRequestTypeChange}
```

Comment:
No need to send onChange handler explicitly in this case. DropdownField handles that.